app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/BrowserNavigationOverlay.kt

```diff
@@ -238,24 +240,26 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
         }
 
         pocketVideoMegaTileView.setOnClickListener(this)
+    }
 
-        if (pocketVideos.isCompleted) {
-            pocketVideoMegaTileView.pocketVideos = pocketVideos.getCompleted()
-
-            if (pocketVideoMegaTileView.pocketVideos == null) {
-                showMegaTileError()
-            } else {
-                pocketVideosContainer.visibility = View.VISIBLE
-                pocketErrorContainer.visibility = View.GONE
-                pocketVideosContainer.contentDescription = resources.getString(R.string.pocket_home_a11y_tile_focused,
-                        resources.getString(R.string.pocket_brand_name))
-            }
-        } else {
-            // TODO: #864 show loading screen
-            launch(UI) {
-                pocketVideoMegaTileView.pocketVideos = pocketVideos.await()
+    fun observeForMegaTile(fragment: Fragment) {
```

Comment:
Why not take a context and a lifecycle owner instead of a fragment?

Reply:
`ViewModelProviders#of` requires either a `Fragment` or `Activity`.